File: functions/node_js_14.x/src/request/validator.ts
import { getPartNumber, getRange } from './utils'
import { UserRequest } from '../s3objectlambda_event'

/**
 * Responsible for validating the user request. Returns a string error message if there are errors or null if valid.
 */
export default function validate (userRequest: UserRequest): string | null {
  const range = getRange(userRequest)
  const partNumber = getPartNumber(userRequest)

  if (range != null && partNumber != null) {
    return 'Cannot specify both Range header and partNumber query parameter'
  }
  return null
}

File: function/python_3_9/src/request/validator.py
from request import utils
from response import range_mapper
from collections import namedtuple

RequestsValidation = namedtuple('RequestsValidation', ['is_valid', 'error_msg'])


def validate_request(user_request):
    """
    Validate the user request
    :param user_request: User request received from AWS Object Lambda
    :return: RequestsValidation Object
    """
    range_request = utils.get_range(user_request)
    part_number_request = utils.get_part_number(user_request)

    if range_request and part_number_request:
        return RequestsValidation(False, 'Cannot specify both Range and Part Number in Query')
    if range_request and not range_mapper.validate_range_str(range_request):
        return RequestsValidation(False, 'Cannot process specific range: {}'.format(range_request))
    return RequestsValidation(True, '')
